apps/daimo-mobile/src/view/shared/AnimatedCheck.tsx

```diff
@@ -0,0 +1,125 @@
+import * as Haptics from "expo-haptics";
+import React, { forwardRef, useImperativeHandle } from "react";
+import { StyleSheet, View } from "react-native";
+import Animated, {
+  useAnimatedProps,
+  useAnimatedStyle,
+  useSharedValue,
+  withDelay,
+  withTiming,
+} from "react-native-reanimated";
+import Svg, { Circle, Path } from "react-native-svg";
+
+import { color } from "./style";
+
+const APath = Animated.createAnimatedComponent(Path);
+const ACircle = Animated.createAnimatedComponent(Circle);
+const ASvg = Animated.createAnimatedComponent(Svg);
+
+const PATH_DASH_OFFSET_INIT = 130;
+const PATH_SCALE_INIT = 1;
+const CIRCLE_DASH_OFFSET_INIT = -380;
+
+export type AnimatedCheckRef = {
+  play: () => void;
+};
+
+export const AnimatedCheck = forwardRef<AnimatedCheckRef>((_, ref) => {
+  const pathDashOffset = useSharedValue(PATH_DASH_OFFSET_INIT);
+  const pathScale = useSharedValue(PATH_SCALE_INIT);
+  const circleDashOffset = useSharedValue(CIRCLE_DASH_OFFSET_INIT);
+
+  const animateSuccess = () => {
+    pathDashOffset.value = PATH_DASH_OFFSET_INIT;
+    circleDashOffset.value = CIRCLE_DASH_OFFSET_INIT;
+    pathScale.value = PATH_SCALE_INIT;
+    setTimeout(() => {
+      Haptics.notificationAsync(Haptics.NotificationFeedbackType.Success);
+      pathDashOffset.value = withDelay(200, withTiming(0, { duration: 500 }));
+      circleDashOffset.value = withTiming(0, { duration: 500 });
+      pathScale.value = withDelay(200, withTiming(0, { duration: 600 }));
+    }, 250);
```

Comment:
i really like how simple and direct the animation setup is

sped it up by about 40%